Sources/CodeEditSourceEditor/Controller/TextViewController+IndentLines.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  TextViewController+IndentLines.swift
+//
+//
+//  Created by Ludwig, Tom on 11.09.24.
+//
+
+import CodeEditTextView
+import AppKit
+
+extension TextViewController {
+    /// Handels indentation and unindentation
+    ///
+    /// Handles the indentation of lines in the text view based on the current indentation option.
+    ///
+    /// This function assumes that the document is formatted according to the current selected indentation option.
+    /// It will not indent a tab character if spaces are selected, and vice versa. Ensure that the document is
+    /// properly formatted before invoking this function.
+    ///
+    /// - Parameter inwards: A Boolean flag indicating whether to outdent (default is `false`).
+    public func handleIndent(inwards: Bool = false) {
+        let indentationChars: String = indentOption.stringValue
+        guard !cursorPositions.isEmpty else { return }
+
+        textView.undoManager?.beginUndoGrouping()
+        for cursorPosition in self.cursorPositions {
+            // get lineindex, i.e line-numbers+1
+            guard let lineIndexes = getHighlightedLines(for: cursorPosition.range) else { continue }
+
+            for lineIndex in lineIndexes {
+                adjustIndentation(
+                    lineIndex: lineIndex,
+                    indentationChars: indentationChars,
+                    inwards: inwards
+                )
+            }
+        }
+        textView.undoManager?.endUndoGrouping()
+    }
+
+    /// This method is used to handle tabs appropriately when multiple lines are selected,
+    /// allowing normal use of tabs.
+    ///
+    /// - Returns: A Boolean value indicating whether multiple lines are highlighted.
+    func multipleLinesHighlighted() -> Bool {
+        for cursorPosition in self.cursorPositions {
+            if let startLineInfo = textView.layoutManager.textLineForOffset(cursorPosition.range.lowerBound),
+               let endLineInfo = textView.layoutManager.textLineForOffset(cursorPosition.range.upperBound),
+               startLineInfo.index != endLineInfo.index {
+                return true
+            }
+        }
+        return false
+    }
+
+    private func getHighlightedLines(for range: NSRange) -> [Int]? {
+        guard let startLineInfo = textView.layoutManager.textLineForOffset(range.lowerBound) else {
+            return nil
+        }
+        var lines: [Int] = [startLineInfo.index]
+
+        guard let endLineInfo = textView.layoutManager.textLineForOffset(range.upperBound),
+              endLineInfo.index != startLineInfo.index else {
+            return lines
+        }
+        if endLineInfo.index == startLineInfo.index + 1 {
+            lines.append(endLineInfo.index)
+            return lines
+        }
+
+        return Array(startLineInfo.index...endLineInfo.index)
```

Comment:
Could this return a range instead of an array? The iterator you have earlier would still work and we'd be a little more explicit about what this is returning. 

Also a small potential perf improvement for massive cursor selections.

Edit: looks like a range won't work but a `ClosedRange` would as its start/end can be equal.

Reply:
Updated the code accordingly